server/service/src/invoice_line/stock_out_line/insert/validate.rs

```diff
@@ -61,6 +63,41 @@ pub fn validate(
     check_location_on_hold(&batch).map_err(|e| match e {
         LocationIsOnHoldError::LocationIsOnHold => LocationIsOnHold,
     })?;
+
+    // If we have an allocated_date older than 24hours, we need to calculate the historical stock line to see if we would have enough stock at that time
+    let batch = if let Some(allocated_date) = invoice.allocated_datetime.clone() {
+        if allocated_date < chrono::Utc::now().naive_utc() - chrono::Duration::hours(24) {
+            let historical_stock_lines =
+                get_historical_stock_lines(ctx, &store_id, &item.id, &allocated_date).map_err(
+                    |e| {
+                        InsertStockOutLineError::DatabaseError(
+                            repository::RepositoryError::DBError {
+                                msg: "Unable to calculate stock levels for this line".to_string(),
+                                extra: format!("{:?}", e),
+                            },
+                        )
+                    },
+                )?;
+            let stockline = historical_stock_lines
```

Comment:
Feels ugly doing this e.g. querying all lines then filtering out in Rust, possibly worth refactoring the  `get_historical_stock_lines` function?
Want to use it to display available stock which needs all lines for the item, but maybe can break it up into smaller functions perhaps?